proxy/src/main/java/com/velocitypowered/proxy/connection/MinecraftConnection.java

```diff
@@ -49,6 +57,13 @@ public MinecraftConnection(Channel channel, VelocityServer server) {
 
     @Override
     public void channelActive(ChannelHandlerContext ctx) throws Exception {
+        if (pluginsSessionHandlers != null && !pluginsSessionHandlers.isEmpty()) {
```

Comment:
I see lines like this and my only reaction is _yuck!_ - this isn't particularly idiomatic.

I'd prefer something more like `firePluginSessionHandlers(handler::connected);`.